src/internal/incompatibility.rs

```diff
@@ -266,7 +266,7 @@ impl<P: Package, V: Version> Incompatibility<P, V> {
     }
 
     /// Iterate over packages.
-    pub fn iter(&self) -> std::collections::hash_map::Iter<P, Term<V>> {
+    pub fn iter(&self) -> indexmap::map::Iter<P, Term<V>> {
```

Comment:
```suggestion
    pub fn iter(&self) -> impl Iterator<Item=(P, Term<V)> {
```

This way we do not commit to the Map type in the API. I dunno though if we commit to the Map type somewhere else.

Reply:
Great suggestion! I remember there were a few similar places I had to change, so I'll include those when doing a solution with another hashing algorithm.

Reply:
I've applied the change, thank you again. The other one I wanted to do it was
`type IntoIter = std::collections::hash_map::IntoIter<P, Term<V>>;`
Doing that in `type` definitions turned out to be unstable :(